tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/HeisenbergOperations.java

```diff
@@ -416,7 +416,7 @@ public String callSaul(@Connection HeisenbergConnection connection) {
   }
 
   @MediaType(TEXT_PLAIN)
-  public String callGusFring() throws HeisenbergException {
+  public String callGusFring(@Connection HeisenbergConnection connection) throws HeisenbergException {
```

Comment:
add new operations/components to do your tests rather than modifying already existing ones that may be testing other stuff

Reply:
This particular operation is throwing a ConnectionException and it was not declaring the handling of a connection, thus not mapping the CONNECTIVITY error and making OperationErrorHandlingTestCase#connectionExceptionThrowsAnConnectivityError() fail after my changes, because the namespace was being resolved to MULE instead of HEISENBERG. I believe that in this case the modification applies, but can be modified to include a new operation if the SDK team thinks that is necessary. Side note: The test passed prior to my changes because the ConnectionException mapping was always added, regardless of the operation declarations.

Reply:
I agree with Rodro here, if you add this you will have to modify any usage of the operation.

How is this operation throwing a ConnectionException?

Reply:
The operation ends up throwing a ConnectionException because of the declaration of the HeisenbergConnectionExceptionEnricher class as an ExceptionHandler of the HeisenbergExtension, transforming any non MuleFatalException into a ConnectionException.